src/Microsoft.ServiceFabric.Services.Queryable/Controller/QueryableController.cs

```diff
@@ -70,6 +74,118 @@ public abstract class QueryableController : ApiController
 			}
 		}
 
+		protected async Task<IHttpActionResult> DeleteAsync(string application, string service, string collection,
+			ValueViewModel[] obj)
+		{
+			var serviceUri = GetServiceUri(application, service);
+			try
+			{
+				Dictionary<Guid, List<JToken>> preMap = new Dictionary<Guid, List<JToken>>();
+
+				Dictionary<Guid, Dictionary<JToken, bool>> parResult = new Dictionary<Guid, Dictionary<JToken, bool>>();
+
+				for (int i = 0; i < obj.Length; i++)
+				{
+					List<JToken> templist = new List<JToken>();
+					if (preMap.ContainsKey(obj[i].PartitionId))
+					{
+						templist = preMap[obj[i].PartitionId];
+						templist.Add(obj[i].Key);
+						preMap[obj[i].PartitionId] = templist;
+					}
+					else
+					{
+						templist.Add(obj[i].Key);
+						preMap[obj[i].PartitionId] = templist;
+					}
+				}
+
+				foreach (Guid mypid in preMap.Keys)
+				{
+					//Fetch partition proxy.
+					var proxy = await GetServiceProxyForDeleteAsync<IQueryableService>(serviceUri, mypid).ConfigureAwait(false);
+					Dictionary<JToken, bool> keyResult = new Dictionary<JToken, bool>();
+
+					foreach (JToken myKey in preMap[mypid])
+					{
+						string keyquoted = JsonConvert.SerializeObject(myKey,
+							new JsonSerializerSettings { StringEscapeHandling = StringEscapeHandling.EscapeNonAscii });
+						keyResult[myKey] = await proxy.DeleteAsync(collection, keyquoted);
+					}
+
+					parResult[mypid] = keyResult;
+				}
+
+				return Ok(parResult);
+			}
+			catch (Exception e)
+			{
+				return HandleException(e, serviceUri);
+			}
+		}
+
+		protected async Task<IHttpActionResult> AddAsync(string application, string service, string collection,
+			ValueViewModel[] obj)
+		{
+			var serviceUri = GetServiceUri(application, service);
+			try
+			{
+				bool[] results = new bool[obj.Length];
+				Dictionary<JToken, bool> results1 = new Dictionary<JToken, bool>();
+
+				for (int i = 0; i < obj.Length; i++)
+				{
+					//Serialize the key from the json body and put it into a string.
+					string keyquoted = JsonConvert.SerializeObject(obj[i].Key,
+						new JsonSerializerSettings { StringEscapeHandling = StringEscapeHandling.EscapeNonAscii });
+					//Serialize the value from the json body and put it into a string.
+					string valuequoted = JsonConvert.SerializeObject(obj[i].Value,
+						new JsonSerializerSettings { StringEscapeHandling = StringEscapeHandling.EscapeNonAscii });
+					//Fetch the proxy
+					var proxy = await GetServiceProxyForAddAsync<IQueryableService>(serviceUri, obj[i].PartitionId)
+						.ConfigureAwait(false);
+					results[i] = await proxy.AddAsync(collection, keyquoted, valuequoted).ConfigureAwait(false);
+					results1[obj[i].Key] = results[i];
+				}
+				return Ok(results1);
```

Comment:
+1 - imo it would be nicer to see the return value as something like:

[
   {
      "PartitionId": <guid>,
      "Key": <key A>,
      "Status":  200
   },
   {
      "PartitionId": <guid>,
      "Key": <key B>,
      "Status":  200
   }
]